webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala

```diff
@@ -59,4 +65,63 @@ final object CheckCoreNumber extends Check {
       case n => Ok(s"${n} cores available")
     }
   }
+}
+
+final object CheckFreeSpace extends Check {
+  def name: CheckName = CheckName("Disk free space available")
+
+  final case class SpaceInfo(val partition: String, val free: Long, val available: Long){
+    def pourcentage: Double =
+      if (available != 0) {
+        (free * 100 / BigDecimal(available)).setScale(2, BigDecimal.RoundingMode.HALF_UP).toDouble
+      } else {
+        0
+      }
+  }
+
+  def run: IOResult[HealthcheckResult] = {
+    val file = root/"proc"/"mounts"
+    val partc = file.lines.map(x => x.split(" ")(1)).toList
+//    val re = "^(/var)".r
+//    val varDir = partc.filter(_.regionMatches(true, 0, "/var", 0, 4))
+//    val spaceTaken = partc.reduceLeft(_ + File(_).size)
+//
+    for {
+      partitionsList <- IOResult.effect{
+        partc.map { x =>
+          val file = new io.File(x)
+          SpaceInfo(x, file.getFreeSpace, file.getTotalSpace)
+        }
+      }
+    } yield {
+      val (varSpaceInfo, rootSpaceInfo) = partitionsList.partition(_.partition.regionMatches(true, 0, "/var", 0, 4))
```

Comment:
I have an hard time followinf what is done, while it's (should be) simple. 
Perhaps can you add comments for the four big steps:
- find partition
- get memory info
- find problems (do math)
- transform to result